src/System/Console/GetOpt/Generics/Internal.hs

```diff
@@ -24,8 +24,13 @@ mapFieldInfo f info = case info of
       cons@Infix{} -> cons
       cons@Constructor{} -> cons
 
-slugify :: String -> String
-slugify [] = []
-slugify (x : xs)
-  | isUpper x = '-' : toLower x : slugify xs
-  | otherwise = x : slugify xs
+normalizeFieldName :: String -> String
+normalizeFieldName =
+    slugify . filter (\ c -> (isAscii c && isAlpha c) || (c == '-'))
```

Comment:
Phew, this triggered a major refactoring of the error handling. But I think it's better now.